controller/mariadb_controller_test.go

```diff
@@ -130,33 +130,33 @@ var _ = Describe("MariaDB controller", func() {
 			}, testTimeout, testInterval).Should(BeTrue())
 
 			By("Expecting to create a exporter Deployment eventually")
-			Eventually(func() bool {
+			Eventually(func(g Gomega) bool {
 				var deploy appsv1.Deployment
 				if err := k8sClient.Get(testCtx, testMariaDb.MetricsKey(), &deploy); err != nil {
 					return false
 				}
 				expectedImage := os.Getenv("RELATED_IMAGE_EXPOTER")
```

Comment:
```suggestion
				expectedImage := os.Getenv("RELATED_IMAGE_EXPORTER")
```

There is a typo here that makes the test fail, as the env variable doesn't exist. Not sure why it was passing before, guessing that using the wrong Gomega instance has implications i.e `g.Expect` vs `Expect` ?

Reply:
Thanks for explaining that. The reason it was passing is the missing assertion of the `Eventually` container. Nothing actually checked this Eventually. 

The different between `Expect` and `g.Expect` is that the first exit Eventually/Constantly immediately, while the second only fails the current loop, letting Eventually to pole again and again until it's eventually working. In most cases, this is what we want.

Reply:
Thanks a lot for the explanation, that makes sense now.